packages/app-spec-squad-chess/src/Board.js

```diff
@@ -47,9 +47,23 @@ function handleSelectPiece() {
     // select the piece and highlight possible turns
     let from = chess.stringToSquare(e.target.id)
     let tos = []
-    state.game.legalTurns.forEach(t => {
-      if (from[0] === t.from[0] &&
-        from[1] === t.from[1]) { tos.push(t.to) }
+    Object.keys(state.game.legalTurns).forEach(turnId => {
+      const turn = state.game.legalTurns[turnId]
+      let match = false
+      // see if the turn involves the from square
+      Object.keys(turn).forEach(s => {
+        const square = chess.stringToSquare(s)
+        if (from[0] === square[0] &&
+          from[1] === square[1]) { match = true }
+      })
+      // if it does, highlight all the other squares from that turn
```

Comment:
This could start failing if there are mechanics that effect a square without a piece actually moving to that square (i.e. capture something up and to the left by moving only up). In that case, this could end up highlighting squares that you can't actually move to.

Reply:
Maybe check the turnIds instead and find matches there.

Reply:
Doubly nested legal turns:
{ fromsquare: {tosquare: diffs, tosquare: diffs}... }